### PR TITLE
[Swift] property, parameter name mapping support

### PR DESCRIPTION
**Pull request by wing328**
- property, parameter name mapping support

tested locally and the diff looks like:
```diff
 internal struct PropertyNameMapping: Codable, JSONEncodable, Hashable {
 
     internal var httpDebugOperation: String?
+    internal var underscoreType: String?
     internal var type: String?
-    internal var type: String?
-    internal var type: String?
-    internal var type: String?
+    internal var typeWithUnderscore: String?
+    internal var dashType: String?
 
-    internal init(httpDebugOperation: String? = nil, type: String? = nil, type: String? = nil, type: String? = nil, type: String? = nil) {
+    internal init(httpDebugOperation: String? = nil, underscoreType: String? = nil, type: String? = nil, typeWithUnderscore: String? = nil, dashType: String? = nil) {
         self.httpDebugOperation = httpDebugOperation
+        self.underscoreType = underscoreType
         self.type = type
-        self.type = type
-        self.type = type
-        self.type = type
+        self.typeWithUnderscore = typeWithUnderscore
+        self.dashType = dashType
     }
 
     internal enum CodingKeys: String, CodingKey, CaseIterable {
         case httpDebugOperation = "http_debug_operation"
-        case type = "_type"
+        case underscoreType = "_type"
         case type
-        case type = "type_"
-        case type = "-type"
+        case typeWithUnderscore = "type_"
+        case dashType = "-type"
     }
 
     // Encodable protocol methods
@@ -39,10 +39,10 @@ internal struct PropertyNameMapping: Codable, JSONEncodable, Hashable {
     internal func encode(to encoder: Encoder) throws {
         var container = encoder.container(keyedBy: CodingKeys.self)
         try container.encodeIfPresent(httpDebugOperation, forKey: .httpDebugOperation)
+        try container.encodeIfPresent(underscoreType, forKey: .underscoreType)
         try container.encodeIfPresent(type, forKey: .type)
-        try container.encodeIfPresent(type, forKey: .type)
-        try container.encodeIfPresent(type, forKey: .type)
-        try container.encodeIfPresent(type, forKey: .type)
+        try container.encodeIfPresent(typeWithUnderscore, forKey: .typeWithUnderscore)
+        try container.encodeIfPresent(dashType, forKey: .dashType)
     }
 }
```

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] In case you are adding a new generator, run the following additional script : 
  ```
  ./bin/utils/ensure-up-to-date
  ``` 
  Commit all changed files.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

FYI @jgavris (2017/07) @ehyche (2017/08) @Edubits (2017/09) @jaz-ah (2017/09) @4brunu (2019/11) @dydus0x14 (2023/06)

### CODE REVIEW

File: bin/configs/swift5-nonPublicApi.yaml [hunk 1]
```diff
@@ -9,4 +9,4 @@ additionalProperties:
   sortParamsByRequiredFlag: false
   nonPublicApi: true
   projectName: PetstoreClient
-  podHomepage: https://github.com/openapitools/openapi-generator
+  podHomepage: https://github.com/openapitools/openapi-generatorq
```
Comment (4brunu) [inline bin/configs/swift5-nonPublicApi.yaml:12]: @wing328 Is this change by accident?

File: samples/client/petstore/swift5/nonPublicApi/PetstoreClient/Classes/OpenAPIs/Models/PropertyNameMapping.swift [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import AnyCodable
+#endif
+
+internal struct PropertyNameMapping: Codable, JSONEncodable, Hashable {
+
+    internal var httpDebugOperation: String?
+    internal var type: String?
+    internal var type: String?
+    internal var type: String?
+    internal var type: String?
```
Comment (4brunu) [inline samples/client/petstore/swift5/nonPublicApi/PetstoreClient/Classes/OpenAPIs/Models/PropertyNameMapping.swift:19]: @wing328 all the properties have the same name, this won't compile

File: bin/configs/swift5-nonPublicApi.yaml (hunk 1 above)
Comment (wing328) [reply bin/configs/swift5-nonPublicApi.yaml:12]: yes, will undo the change

File: samples/client/petstore/swift5/nonPublicApi/PetstoreClient/Classes/OpenAPIs/Models/PropertyNameMapping.swift (hunk 2 above)
Comment (wing328) [reply samples/client/petstore/swift5/nonPublicApi/PetstoreClient/Classes/OpenAPIs/Models/PropertyNameMapping.swift]: yes, this won't compile without using the mapping option, e.g.
```yaml
generatorName: swift5
outputDir: samples/client/petstore/swift5/nonPublicApi
inputSpec: modules/openapi-generator/src/test/resources/2_0/swift/petstore-with-fake-endpoints-models-for-testing.yaml
templateDir: modules/openapi-generator/src/main/resources/swift5
generateAliasAsModel: true
additionalProperties:
  podAuthors: ""
  podSummary: PetstoreClient
  sortParamsByRequiredFlag: false
  nonPublicApi: true
  projectName: PetstoreClient
  podHomepage: https://github.com/openapitools/openapi-generator
nameMappings:
  _type: underscoreType
  type_: typeWithUnderscore
  -type: dashType
parameterNameMappings:
  _type: underscoreType
  type_: typeWithUnderscore
  -type: dashType
```

I removed the files as these files are not supposed to be included. You can refer to the diff in the description to understand the change before and after using nameMappings, parameterNameMappings in the config.

Comment (wing328) [reply samples/client/petstore/swift5/nonPublicApi/PetstoreClient/Classes/OpenAPIs/Models/PropertyNameMapping.swift]: And this is what the schema looks like that will result in naming collisions:
```
components:
  schemas:
    PropertyNameMapping:
      properties:
        http_debug_operation:
          type: string
        _type:
          type: string
        type:
          type: string
        type_:
          type: string
        -type:
          type: string

```

**Pull request merged**